File: app/src/main/java/com/androidexamples/app/activity/PreferenceScreenActivity.java
package com.androidexamples.app.activity;

import android.os.Bundle;
import android.preference.PreferenceFragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.preference.PreferenceFragmentCompat;
import android.support.v7.widget.Toolbar;

import com.androidexamples.app.R;

public class PreferenceScreenActivity extends AppCompatActivity {

    // include 'com.android.support:preference-v7:23.0.1' on gradle
    // no style theme:
    // <item name="preferenceTheme">@style/PreferenceThemeOverlay</item>

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_preference_screen);
        loadComponents();
    }

    private void loadComponents() {
        setSupportActionBar((Toolbar)findViewById(R.id.top_toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.container, new PreferencesFragment());
        fragmentTransaction.commit();
    }

    public static class PreferencesFragment extends PreferenceFragmentCompat {

        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            addPreferencesFromResource(R.xml.preferences);
        }
    }
}

File: app/src/main/java/com/androidexamples/app/flows/PreferenceScreenActivity.kt
package com.androidexamples.app.flows

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.preference.PreferenceFragmentCompat
import android.support.v7.widget.Toolbar
import android.view.View

import com.androidexamples.app.R

class PreferenceScreenActivity : AppCompatActivity() {

    // include 'com.android.support:preference-v7:23.0.1' on gradle
    // no style theme:
    // <item name="preferenceTheme">@style/PreferenceThemeOverlay</item>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_preference_screen)
        loadComponents()
    }

    private fun loadComponents() {
        setSupportActionBar(findViewById<View>(R.id.top_toolbar) as Toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.container, PreferencesFragment())
        fragmentTransaction.commit()
    }

    class PreferencesFragment : PreferenceFragmentCompat() {

        override fun onCreatePreferences(savedInstanceState: Bundle, rootKey: String) {
            addPreferencesFromResource(R.xml.preferences)
        }
    }
}
